src/migrator.js

```diff
@@ -42,18 +44,50 @@ const getTimestampFromFileName = (filename) => {
     return (matches && matches[1]) || null
 }
 
-const getMigrationsToApply = async (space) => {
-    const timestamps = await getMigratedTimestamps(space)
-    return fs.readdirSync(join('.', MIGRATIONS_DIR)).filter((file) => {
-        const timestamp = getTimestampFromFileName(file)
-        return timestamp && !timestamps.includes(timestamp)
+const getNameFromFileName = (filename) => {
+    const matches = filename.match(/^\d{17}-([^.]+)/)
+    return (matches && matches[1]) || null
+}
+
+const getMigrationsToApply = async (space, options) => {
+    const appliedMigrationTimestamps = await getMigratedTimestamps(space)
+
+    let fullMigrations
+    if (options.rollback) {
+        const latestAppliedMigration = fs.readdirSync(join('.', MIGRATIONS_DIR)).find((file) => {
+            const timestamp = getTimestampFromFileName(file)
+            return timestamp && appliedMigrationTimestamps.includes(timestamp)
+        })
+        fullMigrations = latestAppliedMigration ? [latestAppliedMigration] : []
+    } else {
+        fullMigrations = fs.readdirSync(join('.', MIGRATIONS_DIR)).filter((file) => {
+            const timestamp = getTimestampFromFileName(file)
+            return timestamp && !appliedMigrationTimestamps.includes(timestamp)
+        })
+    }
+
+    return fullMigrations.map((f) => {
+        return extractFunctionToSeparateFile(f, options.rollback ? 'down' : 'up')
     })
 }
 
+function extractFunctionToSeparateFile(filePath, direction) {
+    const migrationFile = join('..', MIGRATIONS_DIR, filePath) // TODO Assumes a file structure that might not be correct
```

Comment:
I'm not sure this will work in all cases.

Reply:
I'm actually having some issues with the paths when running the package installed in another project. So we'll have to try that too.